packages/core/src/shareback/create-and-upload-extrinsic-object.ts

```diff
@@ -0,0 +1,95 @@
+import { createExtrinsicObjectXml } from "../external/carequality/dq/create-metadata-xml";
+import { createPatientUniqueId } from "../external/carequality/shared";
+import { S3Utils } from "../external/aws/s3";
+import { DocumentReference, Organization } from "@medplum/fhirtypes";
+import { out } from "../util/log";
+
+const { log } = out("Core Create and Upload Extrinsic Object");
+
+export async function createAndUploadDocumentdMetadataFile({
+  s3Utils,
+  cxId,
+  patientId,
+  docId,
+  size,
+  docRef,
+  metadataFileName,
+  destinationBucket,
+  mimeType,
+}: {
+  s3Utils: S3Utils;
+  cxId: string;
+  patientId: string;
+  docId: string;
+  hash: string;
+  size: string;
+  docRef: DocumentReference;
+  metadataFileName: string;
+  destinationBucket: string;
+  mimeType: string;
+}): Promise<void> {
+  const createdTime = new Date().toISOString();
+  const uniquePatientId = createPatientUniqueId(cxId, patientId);
+
+  const title = docRef?.description;
+  const classCode = docRef?.type;
+  const practiceSettingCode = docRef?.context?.practiceSetting;
+  const healthcareFacilityTypeCode = docRef?.context?.facilityType;
+  const organization: Organization | undefined = docRef?.contained?.find(
+    (resource): resource is Organization => resource.resourceType === "Organization"
+  );
+  log(`Creating metadata file for docId: ${docId}`);
```

Comment:
We should only add meaningful logs so its not too noisy. Maybe we can remove this one?